modal/mount.py

```diff
@@ -130,37 +131,25 @@ def f():
 
     _entries: List[_MountEntry]
 
-    def __init__(
-        self,
-        # Mount path within the container.
-        remote_dir: Union[str, PurePosixPath] = None,
-        *,
-        # Local directory to mount.
-        local_dir: Optional[Union[str, Path]] = None,
-        # Local file to mount, if only a single file needs to be mounted. Note that exactly one of `local_dir` and `local_file` can be provided.
-        local_file: Optional[Union[str, Path]] = None,
-        # Optional predicate to filter files while creating the mount. `condition` is any function that accepts an absolute local file path, and returns `True` if it should be mounted, and `False` otherwise.
-        condition: Optional[Callable[[str], bool]] = None,  # default to include all files
-        # Optional flag to toggle if subdirectories should be mounted recursively.
-        recursive: bool = True,
-        _entries: Optional[List[_MountEntry]] = None,  # internal - don't use
-    ):
+    def __init__(self, *args, **kwargs):
         """The Mount constructor is deprecated. Use static factory method Mount.from_local_dir or Mount.from_local_file"""
-        if _entries is None:
-            deprecation_error(
-                date(2023, 2, 8),
-                self.__init__.__doc__,
-            )
-
-        # TODO(erikbern): remove this code path, and use _from_loader instead
-        assert local_file is None and local_dir is None
-        self._entries = _entries
-        self._is_local = True
-        rep = f"Mount({self._entries})"
-        super().__init__(self._load, rep)
+        deprecation_error(
+            date(2023, 2, 8),
+            self.__init__.__doc__,
+        )
 
-    def extend(self, *entries) -> "_Mount":
-        return _Mount(_entries=[*self._entries, *entries])
+    @staticmethod
+    def _from_entries(*entries: _MountEntry) -> "_Mount":
+        rep = f"Mount({entries})"
+        load = functools.partial(_Mount._load_mount, entries)
+        obj = _Mount._from_loader(load, rep)
+        obj._entries = entries
+        obj._is_local = True
+        return obj
+
+    @property
+    def entries(self):
```

Comment:
Should `entries` be considered a public API? If not, we might want to underscore it to avoid users tripping over it in autocomplete

Reply:
I think it's fine. That being said I think we need to figure out how to store "local" attributes on objects. If you import an existing Mount (either by looking it up, or from inside a container), calling `entries` on it will cause it to fail with an attribute error.